archivebot.py

```diff
@@ -129,14 +129,15 @@ def handle_query(event, cursor, say):
         limit: The number of responses to return. Default 10.
     """
     try:
-        usage_text= "Usage:\n\n\t<query> from:<user> in:<channel> sort:asc|desc limit:<number>\n\n\tquery: The text to search for.\n\tuser: If you want to limit the search to one user, the username. It doesn't work with space separated nicknames\n\tchannel: If you want to limit the search to one channel, the channel name.\n\tsort: Either asc if you want to search starting with the oldest messages, or desc if you want to start from the newest. Default asc.\n\tlimit: The number of responses to return. Default 10."
+        usage_text= "Usage:\n\n\t<query> from:<user> in:<channel> sort:asc|desc limit:<number>\n\n\tquery: The text to search for.\n\tuser: If you want to limit the search to one user, the username. For space separated nicknames, use double quotes in this way '\"from:name surname\" query' \n\tchannel: If you want to limit the search to one channel, the channel name.\n\tsort: Either asc if you want to search starting with the oldest messages, or desc if you want to start from the newest. Default asc.\n\tlimit: The number of responses to return. Default 10."
         text = []
         user_name = None
         channel_name = None
         sort = None
         limit = 10
 
-        params = event["text"].lower().split()
+        s = event["text"].lower()
+        params = ['{}'.format(s).rstrip().lstrip() for s in s.split('"') if s not in ('', ', ')]
```

Comment:
A nice comment explaining what the heck is happening here would be great, even better a very simple function. 😄

Reply:
I added some comment here https://github.com/Sfera-IT/slack-archive-bot/pull/9/commits/e095e95a789bb37ca9c9c6253c0541a542e722ad but it's only a very quick&dirty "hack", If you don't like it I'll try to do something better one day 🤣

Reply:
@esolitos moreover, I added a fallback on "Real Name" if display name is not set https://github.com/Sfera-IT/slack-archive-bot/pull/9/commits/0a84057fd7c59adfa9625b12e4962bebdebdfceb